pkg/transport/queuestorage/file.go

```diff
@@ -141,24 +170,76 @@ func (f *FileWriter) Write(data []byte) (n int, err error) {
 		return
 	}
 
+	f.maxRows--
 	return
 }
 
 func (f *FileWriter) WriteLn(data []byte) (n int, err error) {
 	return f.Write(append(data, '\n'))
```

Comment:
This isn't new code so just a thought... but maybe it's worth fixing this PR?

`append` either modifies the input which is unexpected or it creates a new slice thus wasting memory.

I'd do something like make the write function take a slice of things to write (`[][]byte`), then make `Write` and `WriteLn` wrap that instead.

Reply:
This is fine because we are not modifying any existing value. If `WriteLn()` is changing any item prior to `data[:len(data)-1]` even after append, that is when corruption occurs.

Append will reslice only when there is available capacity or provision a new array with enough space.

Reply:
To clarify: if there is capacity, it will end up modifying the _underlying array_ i.e. the part after the `data` slice, unless `data`'s cap was explicitly set (which we can't guarantee).

e.g. https://go.dev/play/p/twYSbuK0v0U